src/org/exist/collections/Collection.java

```diff
@@ -1774,6 +1774,17 @@ private void manageDocumentInformation(final DocumentImpl oldDoc, final Document
         document.setMetadata(metadata);
     }
 
+     /** Update the modification time of a document
+     * @param document
+     */
+    
+    private void updateModificationTime(final DocumentImpl document) {
+        DocumentMetadata metadata = new DocumentMetadata();
```

Comment:
Also can you make use of `final` unless you variable need to be mutable for some reason please?